### DISCUSSION TITLE
Not Working

### DISCUSSION THREAD

**Discussion by sanigerson** [valinet/ExplorerPatcher#315] [Q&A]
Hi,
I installed latest dll under windows folder, restart explorer.exe, but nothing happens. Also, %appdata%\ExplorerPatcher folder doesn't seems to be created but if i want to delete dll file under windows folder, windows doesn't let me to do it. Any suggestion?
I deactivated AV, also connected to internet.
Thanks a lot.

Windows OS: 22000.318

**Comment by JohnBoy200** [comment]
I've found windows won't allow you to delete the dll but it will let you move it out of windows folder.

**Reply by sanigerson** [reply]
After i rename it, i can delete it. But that's not the problem. After putting it under windows folder, nothing happens.

**Comment by ZyMet** [comment]
Hi,

I had the same issue. reverse back to version 22000.258.26.2 solve the issue for me.

[dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7529057/dxgi.zip)
2

**Reply by sanigerson** [reply]
You, my friend, you are a true hero :)
Thank you very much

**Reply by valinet** [reply]
Reversing to an old version is not a solution, helping debug and finding out what the problem is with the current version is a real solution. Reversing to Windows 10 makes this patch redundant as well, y’know…

**Reply by sanigerson** [reply]
Dear Valinet, 
If there is a way to help you about debug, i am ready. Main reason why i use explorerpatcher is, windows 11 doesn't let me to "never combine" same windows on taskbar. So, Windows10 is already have "explorerpatch" :)

**Reply by valinet** [reply]
Yeah, that’s what I was saying as well. First of all enable the console in Properties. Then collect crash dumps when it crashes. Instructions on how to do that are here: https://github.com/valinet/ExplorerPatcher/issues/320#issuecomment-966469498

Or, idk, stay on old versions, I don’t mind it, I just wished that bugs are eventually identified and taken care of, that’s all. It’s pretty hard to release a new version without maybe breaking things. Updates I push are never problematic on my config, but of course, my setup doesn’t represent the whole variety out there.

**Reply by ZyMet** [reply]
Hi Valinet,

The problem with the latest version is that it won't even start. It doesn't do anything. It won't download the pack. So I can't even open the console, because nothing happen.

**Reply by ZyMet** [reply]
I'm on the latest W11 build with an R9 5900X/X570/RX6700XT. I have 10 other PC's with no problems at all with the latest build

**Reply by valinet** [reply]
Enable the console in old version, then switch to new version. It has to start, okay? Otherwise it would let you delete the file. It starts just fine, that it does not work it’s another thing. With the console you can see how far it goes.

**Reply by ZyMet** [reply]
Error with the library
![Naamloos](https://user-images.githubusercontent.com/94189442/141509309-a663db97-64e9-486c-81a1-711f0996e13d.jpg)

**Reply by valinet** [reply]
No, it’s actually fine, weirdly enough, that’s the code the library should return when properly injected into Start 😂

Anyway, and it just freezes like that? Can you test this version as well, please? Thanks.

[dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7528480/dxgi.zip)

**Comment by ZyMet** [comment]
Not working

![afbeelding](https://user-images.githubusercontent.com/94189442/141510152-8fac0dd3-941b-4cf1-8201-6896b501d2dc.png)

**Reply by valinet** [reply]
Which one? The one in the last message? Why not reply there then?

**Reply by ZyMet** [reply]
Mistake of me. Yes the one you attach

**Reply by valinet** [reply]
No worries. Ok, well, so 26.2 is the absolute latest version that works for you?

**Reply by ZyMet** [reply]
Yes, I have try several version. all the other version crash explorer, The latest doesn't crash explorer, but doesn't load.

**Reply by valinet** [reply]
Well, 26.2 happened a month ago. It’s incredibly old. I can’t believe this breaking change slipped through so many revisions, it’s unbelievable. Idk what else to say, I am speechless, this is beyond disappointment. Do you test with a clean OS? What OS build?

**Reply by sanigerson** [reply]
Valinet, i saw the problem firstly while i was accidently "disabled windows 10 look". Then it never started.
Btw, it's the first time ever i saw "downloading icons" on this old version. So, new version probably never installed icons.

**Reply by ZyMet** [reply]
This is 26.2

This was an upgraded Windows 10 PC. Clean install 3 months ago.

![afbeelding](https://user-images.githubusercontent.com/94189442/141512657-bcd7968f-4c60-4d7d-9fb7-45a9ab327cfd.png)

![afbeelding](https://user-images.githubusercontent.com/94189442/141512690-e113da8f-ed6b-4449-b6ba-5d74b899ac39.png)

**Reply by valinet** [reply]
> Valinet, i saw the problem firstly while i was accidently "disabled windows 10 look". Then it never started.
> Btw, it's the first time ever i saw "downloading icons" on this old version. So, new version probably never installed icons.

Icons? You mean symbols…? What “problem”? I honestly don’t understand what you were saying there.

**Reply by valinet** [reply]
@ZyMet well, it is still disappointing... 26.2 was before the refactoring, before most of a lot of stuff basically... does this release work? https://github.com/valinet/ExplorerPatcher/releases/tag/22000.258.26

Test exactly the DLL in that release, please.

**Reply by ZyMet** [reply]
This one is also working

![afbeelding](https://user-images.githubusercontent.com/94189442/141516677-7497cf7a-2226-4998-9aea-4f2c7458e333.png)

**Reply by valinet** [reply]
Ok, this is the next one: https://github.com/valinet/ExplorerPatcher/releases/tag/22000.258.30. Try with that, please. Thanks.

**Reply by ZyMet** [reply]
258.30 is not working

![afbeelding](https://user-images.githubusercontent.com/94189442/141517523-1d549743-1d7b-4915-91be-3cd94e0a828b.png)

**Reply by ZyMet** [reply]
![afbeelding](https://user-images.githubusercontent.com/94189442/141517768-67a3c39b-4337-4897-821a-0a1c522d748a.png)

**Reply by valinet** [reply]
Okay, give me a second, I will send a new version immediately.

**Reply by valinet** [reply]
Can you test this one please? it's the next commit after 26.3: https://github.com/valinet/ExplorerPatcher/suites/4340889873/artifacts/113822736. Thanks.

**Reply by ZyMet** [reply]
It's not working

![afbeelding](https://user-images.githubusercontent.com/94189442/141531397-46a87254-aee1-46e5-b7ce-92891d38218a.png)

**Reply by valinet** [reply]
Ok, so the breaking change seems to be between https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f and https://github.com/valinet/ExplorerPatcher/commit/453af00138687488f8d7eb20a54e7db9efe5c0c0.

Basically, the automatic build for b2ac6991c22c438f48ee88ee9bd3a7596908050f works: https://github.com/valinet/ExplorerPatcher/suites/4340889873/artifacts/113822736

While the automatic build for 453af00138687488f8d7eb20a54e7db9efe5c0c0 does not work: https://github.com/valinet/ExplorerPatcher/suites/4340944373/artifacts/113824427

Is this correct? Can you confirm? I will start looking for the differences between the 2; one is that the window switcher has been introduced, maybe that interferes somehow.

Thanks

**Reply by ZyMet** [reply]
Build 453af00 works (258.26.3)
Build b2ac699 do not work (258.30.0)

**Reply by valinet** [reply]
b2ac6991c22c438f48ee88ee9bd3a7596908050f without the Windows 10 window switcher:

[dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530357/dxgi.zip)

**Reply by ZyMet** [reply]
I just test it on an other PC. Latest build works fine, until I turn of Windows 10 Taskbar in de settings. After turning of Windows 10 Taskbar in the properties en restart explorer there is no properties anymore in (Windows Button+X)

**Reply by ZyMet** [reply]
> 
> 
> Valinet, i saw the problem firstly while i was accidently "disabled windows 10 look". Then it never started. Btw, it's the first time ever i saw "downloading icons" on this old version. So, new version probably never installed icons.

I think he mean turning of Windows 10 Taskbar

**Reply by ZyMet** [reply]
> 
> 
> [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) without the Windows 10 window switcher:
> 
> [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530357/dxgi.zip)

Not working

**Reply by valinet** [reply]
> 
> 
> > [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) without the Windows 10 window switcher:
> > [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530357/dxgi.zip)
> 
> Not working

What is the setting of "Windows 10 taskbar" when doing these tests? ON or OFF ???

**Reply by ZyMet** [reply]
> 
> 
> > > [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) without the Windows 10 window switcher:
> > > [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530357/dxgi.zip)
> > 
> > 
> > Not working
> 
> What is the setting of "Windows 10 taskbar" when doing these tests? ON or OFF ???

It was on with the other PC. When I turn off Windows 10 taskbar and restart explorer it doesn't work anymore. No properties anymore after that.

**Reply by valinet** [reply]
> 
> 
> > > > [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) without the Windows 10 window switcher:
> > > > [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530357/dxgi.zip)
> > > 
> > > 
> > > Not working
> > 
> > 
> > What is the setting of "Windows 10 taskbar" when doing these tests? ON or OFF ???
> 
> It was on with the other PC. When I turn off Windows 10 taskbar and restart explorer it doesn't work anymore. No properties anymore after that.

I repeat my question. What is the setting of "Windows 10 taskbar" when doing these tests? ON or OFF ??? I can't keep track of what you are saying. You want to help or not? If you want, then follow my instructions, don't work against them. When you did those 2 tests:

> 
> 
> > > > [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) without the Windows 10 window switcher:
> > > > [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530357/dxgi.zip)

What was the setting of "Windows 10 taskbar" when doing these tests? ON or OFF ???

Thanks

**Reply by ZyMet** [reply]
Yesterday it was working fine for me with the latest build. I think I have accidently turn off (Windows 10 taskbar) and after a reboot this morning it doesn't work anymore. The is no properties in the (Windows button+X) menu anymore.

I have test that on 2 other PC now, with the same result.

**Reply by ZyMet** [reply]
Latest build works fine. Until you turn off (Windows 10 taskbar) in de properties. After Turning off (Windows 10 taskbar) it's not working anymore. I can't turn it on anymore, because I can not access the properties anymore.

**Reply by valinet** [reply]
`Win`+`X` menu is untouched when using Windows 11 taskbar. To get back to the Properties GUI, run this command: `rundll32 C:\Windows\dxgi.dll,ZZGUI`.

Taking the above into consideration, can you test the latest version with Windows 10 taskbar enabled on the computer where you run these tests:
> 
> 
> Build [453af00](https://github.com/valinet/ExplorerPatcher/commit/453af00138687488f8d7eb20a54e7db9efe5c0c0) works (258.26.3) Build [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) do not work (258.30.0)

And tell me if it works or not, please? Please do that, and answer that question, please. Thanks.

**Reply by ZyMet** [reply]
> 
> 
> `Win`+`X` menu is untouched when using Windows 11 taskbar. To get back to the Properties GUI, run this command: `rundll32 C:\Windows\dxgi.dll,ZZGUI`.
> 

After reversing back to the latest build and running above command to get the gui and turning on the Windows 10 taskbar it's working again.

**Reply by ZyMet** [reply]
![afbeelding](https://user-images.githubusercontent.com/94189442/141537248-25fb9511-b3b0-456b-82ea-433c117489d3.png)

**Reply by valinet** [reply]
Can you test this, please? 

[dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530543/dxgi.zip)

**Reply by ZyMet** [reply]
> 
> 
> `Win`+`X` menu is untouched when using Windows 11 taskbar. To get back to the Properties GUI, run this command: `rundll32 C:\Windows\dxgi.dll,ZZGUI`.
> 
> Taking the above into consideration, can you test the latest version with Windows 10 taskbar enabled on the computer where you run these tests:
> 
> > Build [453af00](https://github.com/valinet/ExplorerPatcher/commit/453af00138687488f8d7eb20a54e7db9efe5c0c0) works (258.26.3) Build [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) do not work (258.30.0)
> 
> And tell me if it works or not, please? Please do that, and answer that question, please. Thanks.

After turning on the Windows 10 taskbar build b2ac699 (258.30.0) is also working.

**Reply by ZyMet** [reply]
> 
> 
> Can you test this, please?
> 
> [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530543/dxgi.zip)

This one also works

**Reply by valinet** [reply]
Please answer this questions:

1. Does the latest release work with "Windows 10 taskbar" ENABLED? https://github.com/valinet/ExplorerPatcher/releases/tag/22000.318.36.1_e02e4ae
2. Does the latest release work with "Windows 10 taskbar" DISABLED? https://github.com/valinet/ExplorerPatcher/releases/tag/22000.318.36.1_e02e4ae
3. Does this [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530543/dxgi.zip) work with "Windows 10 taskbar" ENABLED?
4. Does this [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530543/dxgi.zip) work with "Windows 10 taskbar" DISABLED?

Test this on the computer where you run these tests: 

> > Build [453af00](https://github.com/valinet/ExplorerPatcher/commit/453af00138687488f8d7eb20a54e7db9efe5c0c0) works (258.26.3) Build [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) do not work (258.30.0)

Please answer THIS QUESTION. Thanks

**Reply by sanigerson** [reply]
> 
> 
> `Win`+`X` menu is untouched when using Windows 11 taskbar. To get back to the Properties GUI, run this command: `rundll32 C:\Windows\dxgi.dll,ZZGUI`.
> 
> Taking the above into consideration, can you test the latest version with Windows 10 taskbar enabled on the computer where you run these tests:
> 
> > Build [453af00](https://github.com/valinet/ExplorerPatcher/commit/453af00138687488f8d7eb20a54e7db9efe5c0c0) works (258.26.3) Build [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) do not work (258.30.0)
> 
> And tell me if it works or not, please? Please do that, and answer that question, please. Thanks.

After restarting ExplorerPatch with your command, I enabled Windows10 Taskbar option and everything works ok for now with latest build. I believe main problem is disabling it. Disable windows 10 taskbar, you'll see that explorerpatch won't start anymore.

**Reply by valinet** [reply]
> 
> 
> Please answer this questions:
> 
>     1. Does the latest release work with "Windows 10 taskbar" ENABLED? https://github.com/valinet/ExplorerPatcher/releases/tag/22000.318.36.1_e02e4ae
> 
>     2. Does the latest release work with "Windows 10 taskbar" DISABLED? https://github.com/valinet/ExplorerPatcher/releases/tag/22000.318.36.1_e02e4ae
> 
>     3. Does this [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530543/dxgi.zip) work with "Windows 10 taskbar" ENABLED?
> 
>     4. Does this [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530543/dxgi.zip) work with "Windows 10 taskbar" DISABLED?
> 
> 
> Test this on the computer where you run these tests:
> 
> > > Build [453af00](https://github.com/valinet/ExplorerPatcher/commit/453af00138687488f8d7eb20a54e7db9efe5c0c0) works (258.26.3) Build [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) do not work (258.30.0)
> 
> Please answer THIS QUESTION. Thanks

@sanigerson Can you do that as well, please? Thanks

**Reply by sanigerson** [reply]
Just a sec.

**Reply by ZyMet** [reply]
> 
> 
> Please answer this questions:
> 
>     1. Does the latest release work with "Windows 10 taskbar" ENABLED? https://github.com/valinet/ExplorerPatcher/releases/tag/22000.318.36.1_e02e4ae
> 
>     2. Does the latest release work with "Windows 10 taskbar" DISABLED? https://github.com/valinet/ExplorerPatcher/releases/tag/22000.318.36.1_e02e4ae
> 
>     3. Does this [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530543/dxgi.zip) work with "Windows 10 taskbar" ENABLED?
> 
>     4. Does this [dxgi.zip](https://github.com/valinet/ExplorerPatcher/files/7530543/dxgi.zip) work with "Windows 10 taskbar" DISABLED?
> 
> 
> Test this on the computer where you run these tests:
> 
> > > Build [453af00](https://github.com/valinet/ExplorerPatcher/commit/453af00138687488f8d7eb20a54e7db9efe5c0c0) works (258.26.3) Build [b2ac699](https://github.com/valinet/ExplorerPatcher/commit/b2ac6991c22c438f48ee88ee9bd3a7596908050f) do not work (258.30.0)
> 
> Please answer THIS QUESTION. Thanks

1. Works
2. Do not works
3. Works
4. Do not Works

**Comment by sanigerson** [comment]
With the old version, i enabled windows 10 taskbar. Then i applied newest one.
1- YES
2- NO
3- YES
4- NO

I realized that, after DISABLING windows10 taskbar, Win+x Properties menu dissapears.

**Reply by valinet** [reply]
Just mentioned this 30 minutes ago:

`Win`+`X` menu is untouched when using Windows 11 taskbar. To get back to the Properties GUI, run this command: `rundll32 C:\Windows\dxgi.dll,ZZGUI`.

In the thread we were all happily chatting. Again, a new thread, I don't understand why...

**Reply by sanigerson** [reply]
Sorry, just tried to be quick.
With that command, everything comes back but there is someting should be done about that problem. I mean, if someone disables windows 10 taskbar, properties goes. It shouldn't :)

**Reply by valinet** [reply]
Yeah, don't worry, take your time, better to keep everything organized. We can chat here as well, no problem...

What can be done...? I don't know how to alter that menu. The command I specified is mentioned in the README as well, only this behavior is not, I will add a mention for it as well. This tool was originally designed to bring back the Windows 10 taskbar, so the entire "disable Windows 10 taskbar" is a bit weird, but I decided to support that as well considering that today there is a lot more stuff this tool does compared to what it supported originally, and maybe some people still want to use it but keep the Windows 11 taskbar.

Idk, what else would you propose?

**Reply by ZyMet** [reply]
> 
> 
> Just mentioned this 30 minutes ago:
> 
> `Win`+`X` menu is untouched when using Windows 11 taskbar. To get back to the Properties GUI, run this command: `rundll32 C:\Windows\dxgi.dll,ZZGUI`.
> 
> In the thread we were all happily chatting. Again, a new thread, I don't understand why...

It seems the issue is when you turning off Windows 10 taskbar the GUI is not accessable through Win+X. You have to run the command to access the GUI and turning on Windows 10 taskbar.

**Reply by ZyMet** [reply]
> 
> 
> Yeah, don't worry, take your time, better to keep everything organized. We can chat here as well, no problem...
> 
> What can be done...? I don't know how to alter that menu. The command I specified is mentioned in the README as well, only this behavior is not, I will add a mention for it as well. This tool was originally designed to bring back the Windows 10 taskbar, so the entire "disable Windows 10 taskbar" is a bit weird, but I decided to support that as well considering that today there is a lot more stuff this tool does compared to what it supported originally, and maybe some people still want to use it but keep the Windows 11 taskbar.
> 
> Idk, what else would you propose?

I am very happy with your tool. It makes my PC a lot better. Thnx for your hard works.

**Reply by sanigerson** [reply]
> 
> 
> Yeah, don't worry, take your time, better to keep everything organized. We can chat here as well, no problem...
> 
> What can be done...? I don't know how to alter that menu. The command I specified is mentioned in the README as well, only this behavior is not, I will add a mention for it as well. This tool was originally designed to bring back the Windows 10 taskbar, so the entire "disable Windows 10 taskbar" is a bit weird, but I decided to support that as well considering that today there is a lot more stuff this tool does compared to what it supported originally, and maybe some people still want to use it but keep the Windows 11 taskbar.
> 
> Idk, what else would you propose?

Your effort is priceless. Thank you very much.
I can suggest to add a sentence next to "disable windows 10 taskbar" that it will remove properties and that command should run to restart explorerpatch.

**Reply by valinet** [reply]
@ZyMet  Ok. So basically you went back to the build that did not account for this disable Windows 10 taskbar, that is why it was working with 26.2, that's the meaning behind all of this.

I'd say the problem is ExplorerPatcher does not currently run properly when "enable Windows 10 taskbar" is OFF. It should work with the Windows 11 taskbar as well. Idk, I have to find a fix for that. Tried yesterday, in 36.1, it seems it does not fully work. I will try some stuff again, I don't understand why doing less stuff breaks it, though =))))

**Reply by ZyMet** [reply]
> 
> 
> @ZyMet Ok. So basically you went back to the build that did not account for this disable Windows 10 taskbar, that is why it was working with 26.2, that's the meaning behind all of this.
> 
> I'd say the problem is ExplorerPatcher does not currently run properly when "enable Windows 10 taskbar" is OFF. It should work with the Windows 11 taskbar as well. Idk, I have to find a fix for that. Tried yesterday, in 36.1, it seems it does not fully work. I will try some stuff again, I don't understand why doing less stuff breaks it, though =))))

I am on the latest build now. Everything works fine now with W10 taskbar ON

**Reply by valinet** [reply]
@sanigerson I will think about that. It's also hard to do everything at once, I knew about this issue with people not knowing how to get back to "properties" after disabling Windows 10 taskbar, but I just haven't yet had the time to develop a proper fix, that's all.

**Reply by valinet** [reply]
@ZyMet Great. I will come back with an update to hopefully address the situation when Windows 11 taskbar is enabled as well.

**Reply by sanigerson** [reply]
@valinet I'm very thankfull to you. Again, thank you for time 🥇

**Reply by ZyMet** [reply]
> 
> 
> @ZyMet Great. I will come back with an update to hopefully address the situation when Windows 11 taskbar is enabled as well.

Great. We now know how to adress the issue. Keep up your great works and thnx again. It's my bedtime now. Cheers.

**Reply by valinet** [reply]
I was thinking, maybe I could change "Taskbar settings" to open EP Properties. I know how to do that. That is available in the context menus of both Windows 10 and Windows 11 taskbar. Then provide a link in EP to open the Windows taskbar settings in Settings. That way, people get the same behavior no matter which taskbar they use... and optionally still show a "Properties" entry in `Win`+`X` for people who got used to that...? Idk...

**Reply by ZyMet** [reply]
> 
> 
> I was thinking, maybe I could change "Taskbar settings" to open EP Properties. I know how to do that. That is available in the context menus of both Windows 10 and Windows 11 taskbar. Then provide a link in EP to open the Windows taskbar settings in Settings. That way, people get the same behavior no matter which taskbar they use... and optionally still show a "Properties" entry in `Win`+`X` for people who got used to that...? Idk...

Sound good to me.

**Comment by ZyMet** [comment]
I just send you a beer (donations)
